### ISSUE TITLE
Project does not retain code style settings

### ISSUE THREAD

**Issue by imrekoszo** [cursive-ide/cursive#2681]
I have my default settings but for certain projects I want to use something else. This works if I open the project, set up project-wide code style settings but is not retained when the project is closed then reopened.

```
IntelliJ IDEA 2021.3.3 (Community Edition)
Build #IC-213.7172.25, built on March 15, 2022
Runtime version: 11.0.14.1+1-b1751.46 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 12.3.1
GC: G1 Young Generation, G1 Old Generation
Memory: 4096M
Cores: 8
Registry:
    scala.erase.compiler.process.jdk.once=false

Non-Bundled Plugins:
    org.intellij.plugins.hcl (0.7.15)
    net.seesharpsoft.intellij.plugins.csv (2.18.2)
    name.kropp.intellij.makefile (213.5744.224)
    de.femtopedia.diffplugin (2.1.7)
    com.jetbrains.plugins.ini4idea (213.5744.190)
    org.jetbrains.kotlin (213-1.6.21-release-334-IJ6777.52)
    org.jetbrains.idea.grammar (2021.1.2)
    gherkin (213.6461.19)
    dev.monogon.cuelang (0.8.4)
    org.asciidoctor.intellij.asciidoc (0.37.15)
    StringToolsPlugin (4.0)
    String Manipulation (9.4.0)
    Key Promoter X (2021.3)
    mobi.hsz.idea.gitignore (4.4.0)
    com.cursiveclojure.cursive (1.12.3-eap1-2021.3)
    com.github.shinichy.integrant (0.1.2-imrekoszo)
    com.github.brcosta.cljstuffplugin (0.6.2-imrekoszo)
    EDN-JSON Converter (0.0.5)
    org.intellij.scala (2021.3.20)

Kotlin: 213-1.6.21-release-334-IJ6777.52
```

Repro:

1. verify default settings
![step1](https://user-images.githubusercontent.com/1116295/164181202-727b3a52-3ef1-4d60-ab28-0abaeec04f25.png)
2. `git clone -o upstream git@github.com:lambdaisland/kaocha-cloverage.git`
3. open it as a deps.edn project
4. verify default settings work
![step4](https://user-images.githubusercontent.com/1116295/164181209-c7894d52-50f4-4e8d-9087-1a187338bd95.png)
5. set project-wide settings
![step5](https://user-images.githubusercontent.com/1116295/164181213-d954b04a-37ec-4d0f-8e3b-a0d9bb89d641.png)
6. verify they work
![step6](https://user-images.githubusercontent.com/1116295/164181215-d71ffffb-949a-430b-96fa-08bac98181a1.png)
7. close and reopen project
8. observe global formatting rules are in effect
![step8](https://user-images.githubusercontent.com/1116295/164181217-385bc9ae-cfb1-42b4-8b35-b79528395385.png)
9. check settings, observe Default IDE is selected
![step9](https://user-images.githubusercontent.com/1116295/164181218-393c267e-1a26-4f2b-a82e-62675a2ca7ea.png)
10. select Project, observe settings are not where I had set them
![step10](https://user-images.githubusercontent.com/1116295/164181221-091ac807-0b05-4742-bec5-03dd669caaa8.png)

**Comment by imrekoszo** [comment]
Update: I found that if I have multiple projects open in idea and set this for one of them, all the others will also be affected, i.e. the ones I want to keep using the globals for will also receive the project-specific settings while they are open.

**Comment by imrekoszo** [comment]
I retried this with 2022.2 on a project I had had this problem with and it persisted (project was [created in Apri 2022](https://github.com/imrekoszo/polylith-kaocha/commit/c9cd659015addeaf0fd6a4506cce0b9fbb285459#diff-6efe4c1624e0c57480cdae46d18c16fdca93a234d8e1e50531948155ff76008b)).
Tried with one that I only imported to IJ a few days ago and it worked as expected.

Went back to the problematic project, deleted .idea and reimported the project, which resulted in a diff containing (among others):

<img width="311" alt="image" src="https://user-images.githubusercontent.com/1116295/182348393-bbb6008e-1ce3-49e6-8ea9-f465e3d86047.png">

Now it works as expected. At some point there must have been a migration from codeStyles being a file to it being a folder that didn’t seem to get to these projects.

@cursive-ide mystery solved for me but I'm leaving the issue open in case you might think this is some project migration bug. Feel free to close it if you consider this something Cursive doesn't have an effect on.
